### PR TITLE
[Part 2] Port Auto link 

### PR DESCRIPTION
**Pull request by juliaroldi**
This change is part of the effort to port the Auto Link from the legacy ContentEdit feature into `AutoFormatPlugin`.  After this change `AutoFormatPlugin` will automatically transform a typed url when space key is pressed after the text segment.
AutoLinkTyping

### CODE REVIEW

File: packages/roosterjs-content-model-plugins/lib/autoFormat/link/createLink.ts [hunk 1]
```diff
@@ -5,7 +5,10 @@ import type { IEditor } from 'roosterjs-content-model-types';
 /**
  * @internal
  */
-export function createLink(editor: IEditor) {
+export function createLink(editor: IEditor, autoLink: boolean) {
+    if (!autoLink) {
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model-plugins/lib/autoFormat/link/createLink.ts:9]: same here

File: packages/roosterjs-content-model-plugins/lib/autoFormat/list/keyboardListTrigger.ts [hunk 2]
```diff
@@ -13,26 +13,28 @@ export function keyboardListTrigger(
...
-        if (listStyleType) {
-            const segmentsAndParagraphs = getSelectedSegmentsAndParagraphs(model, false);
-            if (segmentsAndParagraphs[0] && segmentsAndParagraphs[0][1]) {
-                segmentsAndParagraphs[0][1].segments.splice(0, 1);
-            }
-            const { listType, styleType, index } = listStyleType;
-            triggerList(editor, model, listType, styleType, index);
-            rawEvent.preventDefault();
-            normalizeContentModel(model);
+    if (shouldSearchForBullet || shouldSearchForNumbering) {
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model-plugins/lib/autoFormat/list/keyboardListTrigger.ts:16]: same here, check outside to reduce unnecessary parameters.

Comment (juliaroldi) [reply packages/roosterjs-content-model-plugins/lib/autoFormat/list/keyboardListTrigger.ts:16]: We already use these parameter on line 20 and 21, then we cannot remove them

Comment (juliaroldi) [reply packages/roosterjs-content-model-plugins/lib/autoFormat/list/keyboardListTrigger.ts:16]: Should I return this check to outside the function anyway?

Comment (JiuqingSong) [reply packages/roosterjs-content-model-plugins/lib/autoFormat/list/keyboardListTrigger.ts:16]: I see. In that case we can keep it

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**